### PR TITLE
DataFetchResult generaton

### PR DESCRIPTION
**Pull request by kirill071992**
---

### Description

Related to #1416

---

Changes were made to:
- Codegen library - Java
- Gradle plugin

### CODE REVIEW

File: docs/codegen-options.md [hunk 1]
```diff
@@ -1,75 +1,76 @@
...
+|                      `typesAsInterfaces`                       |                              Set(String)                               |                             Empty                              | Types that must generated as interfaces should be defined here in format: `TypeName` or `@directive`. E.g.: `User`, `@asInterface`.                                                                                                                                                                                                                                            |
+|                 `fieldsWithDataFetcherResult`                  |                              Set(String)                               |                             Empty                              | Types that must generated as interfaces should be defined here in format: `TypeName` or `@directive`. E.g.: `User`, `@dataFetcherResult`.                                                                                                                                                                                                                                            |
```
Comment (kobylynskyi) [inline docs/codegen-options.md:68]: @kirill071992 description is incorrect

File: src/main/java/com/kobylynskyi/graphql/codegen/mapper/GraphQLTypeMapper.java [hunk 2]
```diff
@@ -4,6 +4,7 @@
 import com.kobylynskyi.graphql.codegen.model.NamedDefinition;
 import com.kobylynskyi.graphql.codegen.model.RelayConfig;
 import com.kobylynskyi.graphql.codegen.model.definitions.ExtendedFieldDefinition;
+import graphql.com.google.common.collect.ImmutableList;
```
Comment (kobylynskyi) [inline src/main/java/com/kobylynskyi/graphql/codegen/mapper/GraphQLTypeMapper.java:7]: Unused import.

**Comment by kobylynskyi** [comment]
@kirill071992 I have a question about the overall approach. What if user wants to specify a lsit of types or specific fields that should be generated with `DataFetcherResult<>`? Currently you support it only through directives, which is not flexible.

**Review by kobylynskyi — CHANGES_REQUESTED** [review]

**Comment by kirill071992** [comment]
> @kirill071992 I have a question about the overall approach. What if user wants to specify a lsit of types or specific fields that should be generated with `DataFetcherResult<>`? Currently you support it only through directives, which is not flexible. All other config options do support other options besides `@directive`, which are: `Type`, or `Type.field`. Please follow similar logic. Thanks

Done

**Comment by kirill071992** [comment]
@kobylynskyi  :)

File: src/main/java/com/kobylynskyi/graphql/codegen/model/MappingConfig.java [hunk 3]
```diff
@@ -606,6 +608,15 @@ public Set<String> getFieldsToExcludeFromGeneration() {
         return fieldsToExcludeFromGeneration;
     }
 
+    public void setFieldsWithDataFetcherResult(Set<String> fieldsWithDataFetcherResult) {
```
Comment (kobylynskyi) [inline src/main/java/com/kobylynskyi/graphql/codegen/model/MappingConfig.java:611]: Please fix the order of getter/setter. Should not interfere with other pairs.

**Review by kobylynskyi — CHANGES_REQUESTED** [review]
@kirill071992 looks great! One minor comment and we should be good to go. Thanks!

File: src/main/java/com/kobylynskyi/graphql/codegen/model/MappingConfig.java (hunk 3 above)
Comment (kirill071992) [reply src/main/java/com/kobylynskyi/graphql/codegen/model/MappingConfig.java]: @kobylynskyi done :)

**Review by kobylynskyi — APPROVED** [review]
@kirill071992 looks good! Will merge soon and include in the next release.
Thanks for the contribution!

**Review by kobylynskyi — APPROVED** [review]

**Comment by kobylynskyi** [comment]
This was released in 5.10.0. Thanks!

**Pull request merged**